### ISSUE TITLE
exchangeLight.sol questions

### ISSUE THREAD

**Issue by gundas** [CORIONplatform/solidity#157]
Hello,

I've started looking at _exchangeLight.sol_. I see that in `receiveToken(...)` there is a check that the Ether balance of the receiver is less than 0.2 ETC. What is the point? If I really wanted, I could create automated scripts which would empty Ether from the account every time before transferring the COR to the exchange?

Another thing:
```
        uint256 _amount = amount;
        if ( _max > _amount ) {
            _amount = _max;
        }
```
will always make sure that `_amount` is **more** or equals to `amount` (and `_max`).
Then this will result in negative value:
```
return ( true, safeSub(amount, _amount) );
```

**Comment by iFA88** [comment]
Yeah you can write scripts for this to buy and buy it again, but that will consume so much gas that not worth it. We want this for covering gas consume and not for sell tokens.

In the if is a typo! This should be, if the `amount` is greater than the `max`, then should be `max`.

**Comment by gundas** [comment]
How much gas does the Token.transfer(...) consume?  1.000.000 gas could cost around 0.30 USD on ETC.
0.2 ETC = 3 USD.

I still think it will be viable for someone to use this light exchange to convert all of his COR to ETC.

**Comment by iFA88** [comment]
COR transaction cost 150k - 400k GAS.

We will discus of that.
